allel/util.py

```diff
@@ -145,6 +146,50 @@ def ensure_square(dist):
     return dist
 
 
+def mask_inaccessible(is_accessible, ac, pos, start=None, stop=None):
+    """
+    This function returns a tuple (allele counts array, positions) in which
+    positions that are not accessible are removed from the allele counts
+    array and the vector of variant positions.
+
+    Parameters
+    ----------
+    is_accessible : array_like, bool, shape (len(contig),)
+        Boolean array indicating accessibility status for all positions in the
+        chromosome/contig.
+    ac : array_like, int, shape (n_variants, n_alleles)
+        Allele counts array.
+    pos : array_like, int, shape (n_variants,)
+        Variant positions, using 1-based coordinates, in ascending order.
+    start : int, optional
+        The position at which to start (1-based).
+    stop : int, optional
+        The position at which to stop (1-based).
+
+    Returns
+    -------
+    ac : array_like, int, shape (n_variants, n_alleles)
+        Allele counts array consisting exclusively of accessible sites in the
+        original allele counts array.
+    pos : array_like, int, shape (n_items,)
+        Positions array consisting exclusively of accessible sites in the
+        original positions array.
+
+    """
+
+    if is_accessible is not None:
+        # sanity check
+        if np.max(pos) > len(is_accessible):
+            raise ValueError(
+                'Not all positions are covered by is_accessible'
+                )
+        if np.any(np.logical_not(is_accessible[pos-1])):
+            warnings.warn("Some variants were inaccessible (and hence masked)")
+            ac = ac[is_accessible[pos-1]]
+            pos = pos[is_accessible[pos-1]]
```

Comment:
```suggestion
        loc_accessible = is_accessible[pos-1]
        if np.any(np.logical_not(loc_accessible)):
            warnings.warn("Some variants were inaccessible (and hence masked)")
            ac = ac[loc_accessible]
            pos = pos[loc_accessible]
```

Very minor optimisation, avoid computing the mask multiple times.